src/ar_http_iface_h.erl

```diff
@@ -8,22 +8,31 @@
 %%%===================================================================
 
 init(InitialReq, State) ->
-	{Status, Headers, Body, HandledReq} = do_handle(InitialReq),
-	CowboyStatus = handle208(Status),
-	RepliedReq = cowboy_req:reply(CowboyStatus, Headers, Body, HandledReq),
-	{ok, RepliedReq, State}.
+	Pid = self(),
+	spawn_link(fun() ->
+		Pid ! handle(InitialReq)
+	end),
+	receive
+		{Status, Headers, Body, HandledReq} ->
+			CowboyStatus = handle208(Status),
+			RepliedReq = cowboy_req:reply(CowboyStatus, Headers, Body, HandledReq),
+			{ok, RepliedReq, State}
+	after
+		30000 ->
+			error(handler_timeout)
```

Comment:
This is expected to happen, so I don't think we should call `error/1` and cause an error log entry. Instead I think we should reply with a 500 and call `ar:info` or `ar:warn`.

Reply:
Will do 👍